src/sync/SSEHandler/NotificationKeeper.js

```diff
@@ -0,0 +1,56 @@
+import { PUSH_CONNECT, PUSH_DISCONNECT, PUSH_DISABLED, ControlTypes } from '../constants';
+
+const controlPriMatcher = /control_pri$/;
+
+export default function notificationKeeperFactory(feedbackLoopEmitter) {
+
+  let isStreamingUp;
+  let occupancyTimestamp = -1;
+  let controlTimestamp = -1;
+
+  return {
+    handleOpen() {
+      isStreamingUp = true;
+      feedbackLoopEmitter.emit(PUSH_CONNECT);
+    },
+
+    handleOccupancyEvent(publishers, channel, timestamp) {
+      if (controlPriMatcher.test(channel) && timestamp > occupancyTimestamp) {
+        occupancyTimestamp = timestamp;
+        if (publishers === 0 && isStreamingUp) {
+          isStreamingUp = false;
+          feedbackLoopEmitter.emit(PUSH_DISCONNECT); // notify(STREAMING_DOWN) in spec
+          return;
```

Comment:
If you use an else before the next if (keep the if too as I think it's needed) you don't need to have this return. I don't know if the sonarqube configured today tracks this as an increase in complexity but still, easy to avoid. 

FYI only asking this as there's nothing else following the if on line 25. If the function execution would continue further then the return is totally fine.